modifiers/allocation/modifier.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+from enum import Enum
+from ramble.modkit import *
+
+
+class AllocOpt(Enum):
+    # Experiment resource requests
+    N_RANKS = 1
+    N_NODES = 2
+    N_CORES_PER_TASK = 3
+    N_THREADS = 4  # number of OMP threads per task
+    N_RANKS_PER_NODE = 5
+    N_GPUS = 6
+
+    # Descriptions of resources available on systems
+    GPUS_PER_NODE = 100
+    CPUS_PER_NODE = 101
+
+    # Scheduler identification and other high-level instructions
+    SCHEDULER = 200
```

Comment:
I think you are using some sort of numbering system here.  How does this work?

Reply:
These are enum codes. There's nothing required about them other than that they are distinct. I also broadly divide them into categories for ease of reading.